common/web/lm-worker/src/model-compositor.ts

```diff
@@ -160,12 +136,15 @@ class ModelCompositor {
       // The 'eventual' logic will be significantly more complex, though still manageable.
       let searchSpace = contextState.searchSpace[0];
 
-      let newEmptyToken = false;
+      let newToken = false;
       // Detect if we're starting a new context state.
       let contextTokens = contextState.tokens;
       if(contextTokens.length == 0 || contextTokens[contextTokens.length - 1].isNew) {
-        if(this.isEmpty(inputTransform) || this.isWhitespace(inputTransform)) {
-          newEmptyToken = true;
+        // Always note if we have a new token (so that we don't try to delete existing context)
+        newToken = true;
+        // If the new token is due to whitespace, or if we had a context-reset trigger this (thus, no input...)
+        // (Lingering question:  do we need the .isEmpty check here?  Track `prefixTransform` and find out.)
```

Comment:
Is this a TODO? Do we need to sort this out now? Can we mark it as a TODO if we don't need to sort it out now?